tests/simulator/__init__.py

```
@@ -308,3 +313,33 @@ def clean_up_diags_dirs(self):
                     shutil.rmtree(diag_dir)
         cpp_lib().mpi_barrier()
+
+
+def debug_tracer():
+    """
+    print live stack trace during execution
+    """
+
+    import sys, os
+
+    def tracefunc(frame, event, arg, indent=[0]):
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)